Mathlib/Data/Set/Defs.lean

```diff
@@ -0,0 +1,345 @@
+/-
+Copyright (c) 2016 Jeremy Avigad. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Jeremy Avigad, Johannes Hölzl, Reid Barton, Scott Morrison, Patrick Massot, Kyle Miller,
+Minchao Wu, Yury Kudryashov, Floris van Doorn
+-/
+import Std.Classes.SetNotation
+import Mathlib.Order.Basic
+import Mathlib.Util.CompileInductive
+import Mathlib.Data.SProd
```

Comment:
Just checking: is this a weaker set of imports than `Mathlib.Data.Set.Basic` originally had. If so, great!

If not (i.e. you've had to add imports from some later file than `Mathlib.Data.Set.Basic` in order to move definitions from that file to here), then I'd like to know exactly what was added.

Reply:
`Mathlib.Data.SProd` and its dependencies (see below) are new `import`s. Do you want me to move `Set.prod` back to `Data.Set.Prod`?
```dot
digraph "import_graph" {
  "Mathlib.Tactic.DeriveToExpr" -> "Mathlib.Tactic.ToExpr";
  "Mathlib.Util.WhatsNew" -> "Mathlib.Tactic.ToExpr";
  "Mathlib.Tactic.FBinop" -> "Mathlib.Data.SProd";
  "Mathlib.Tactic.ToLevel" -> "Mathlib.Tactic.DeriveToExpr";
  "Mathlib.Mathport.Rename" -> "Mathlib.Tactic.ToLevel";
  "Mathlib.Tactic.PPWithUniv" -> "Mathlib.Tactic.ToLevel";
  "Mathlib.Tactic.ToExpr" -> "Mathlib.Tactic.FBinop";
}
```

Reply:
No, those ones are no problem at all. Please proceed!